Core/URLExtension.swift

```diff
@@ -20,8 +20,15 @@
 
 import Foundation
 
+
+
 extension URL {
     
+    enum URLProtocol: String {
+        case http
+        case https
+    }
+    
     private static let webUrlRegex = "^(https?:\\/\\/)?([\\da-z\\.-]+\\.[a-z\\.]{2,6}|(([\\d]+[.]){3}[\\d]+))([\\/]?[\\/:?=&#]{1}[\\%\\da-zA-Z_\\.-]+)*[\\/\\?]?$"
```

Comment:
I liked that idea and went away and did it but now that I see it I think it obscures the meaning of the regex as `https?` means `http` or `https`